### PR TITLE
nvidia_deeprecommender: benchmark coverage for custom devices

### PR DESCRIPTION
**Pull request by weishi-deng**
Works for Roadmap https://github.com/pytorch/benchmark/issues/1293 to increase benchmark coverage.

For this model, when running on the custom devices except for CPU and CUDA(e.g. XPU), it will raise the error "'DeepRecommenderTrainBenchmark' object has no attribute 'rencoder'" as it will not initialize the rencoder for the custom devices.
In this PR, we accept the device args as a param within the training process and inference process which will cover the model initializing and data transposition for these custom devices.

### CODE REVIEW

File: torchbenchmark/models/nvidia_deeprecommender/nvinfer.py [hunk 1]
```diff
@@ -93,20 +94,24 @@ def processArgState(args) :
       quit()
   
   args.use_cuda = torch.cuda.is_available() # global flag
+  args.use_xpu = args.device == 'xpu'
```
Comment (EikanWang) [inline torchbenchmark/models/nvidia_deeprecommender/nvinfer.py:97]:
```suggestion
  args.use_xpu = torch.xpu.is_available()
```

File: torchbenchmark/models/nvidia_deeprecommender/nvinfer.py [hunk 2]
```diff
@@ -93,20 +94,24 @@ def processArgState(args) :
...
       print('GPU is available.') 
     else: 
       print('GPU is not available.')
   
   if args.use_cuda and args.forcecpu:
       args.use_cuda = False
   
+  if args.use_xpu and args.forcecpu:
+      args.use_xpu = False
+
```
Comment (EikanWang) [inline torchbenchmark/models/nvidia_deeprecommender/nvinfer.py:109]:
```suggestion
  if args.forcecpu:
      args.use_cuda = False
      args.use_xpu = False

```

File: torchbenchmark/models/nvidia_deeprecommender/nvtrain.py [hunk 3]
```diff
@@ -120,8 +120,9 @@ def processTrainArgState(args) :
     quit()
 
   args.use_cuda = torch.cuda.is_available() # global flag
+  args.use_xpu = args.device == 'xpu'
   if not args.silent:
-    if args.use_cuda:
```
Comment (EikanWang) [inline torchbenchmark/models/nvidia_deeprecommender/nvtrain.py:124]: Is this model CUDA-specific as its name?

File: torchbenchmark/models/nvidia_deeprecommender/nvinfer.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ def getCommandLineArgs() :
 
   return args
 
-def getBenchmarkArgs(forceCuda):
```
Comment (EikanWang) [inline torchbenchmark/models/nvidia_deeprecommender/nvinfer.py:60]: Is this model cuda-specific?

File: torchbenchmark/models/nvidia_deeprecommender/nvtrain.py (hunk 3 above)
Comment (weishi-deng) [reply torchbenchmark/models/nvidia_deeprecommender/nvtrain.py:124]: Yes, it's a model from one of the research projects of Nvidia. https://github.com/NVIDIA/DeepRecommender.

File: torchbenchmark/models/nvidia_deeprecommender/nvinfer.py (hunk 4 above)
Comment (weishi-deng) [reply torchbenchmark/models/nvidia_deeprecommender/nvinfer.py:60]: Same as the above comments.

**Pull request closed without merging**